docs/content/guides/rows/row-sorting.md

```diff
@@ -453,163 +1238,919 @@ import 'handsontable/dist/handsontable.full.min.css';
+        price: 314,
+        sellDate: '27/03/2020',
+        sellTime: '51-6775',
+        inStock: true,
+      },
+      {
+        brand: 'Buzzdog',
+        model: 'Mountain Frame',
+        price: 268,
+        sellDate: '29/08/2020',
+        sellTime: '44-4028',
+        inStock: false,
+      },
+      {
+        brand: 'Katz',
+        model: 'Half-Finger Gloves',
+        price: 204,
+        sellDate: '02/10/2020',
+        sellTime: '08-2758',
+        inStock: true,
+      },
+    ]}
+      columns={[
+        {
+          type: 'text',
+          data: 'brand',
+        },
+        {
+          type: 'text',
+          data: 'model',
+        },
+        {
+          type: 'numeric',
+          data: 'price',
+          numericFormat: {
+            pattern: '$ 0,0.00',
+            culture: 'en-US'
+          },
+        },
+        {
+          type: 'date',
+          data: 'sellDate',
+        },
+        {
+          type: 'text',
+          data: 'sellTime',
+        },
+        {
+          type: 'checkbox',
+          data: 'inStock',
+        },
+      ]}
+      colHeaders={[
+        'Brand',
+        'Model',
+        'Price',
+        'Date',
+        'Time',
+        'In stock',
+      ]}
+      columnSorting={{
+        // at initialization, sort rows by column 1, in descending order
+        initialConfig: {
+          column: 1,
+          sortOrder: 'desc',
+        },
+      }}
+      className="your-handsontable-instance"
+      rowHeaders={true}
+      height="auto"
+      width="auto"
+      licenseKey="non-commercial-and-evaluation"
+    />
+  );
+};
+
+/* start:skip-in-preview */
+ReactDOM.render(<YourHandsontableComponent />, document.getElementById('example5'));
+/* end:skip-in-preview */
+```
+:::
+
+:::
+
+## Add a custom comparator
+
+To implement your own [compare function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#description), use the [`compareFunctionFactory`](@/api/options.md#columnsorting) option.
+
+::: only-for javascript
+
+```js
+const configurationOptions = {
+  columnSorting: {
+    compareFunctionFactory(sortOrder, columnMeta) {
+      return function(value, nextValue) {
+        // here, add a compare function
+        // that returns `-1`, or `0`, or `1`
+      }
+    },
+  },
+};
+```
+
 :::
 
 ::: only-for react
+
 ```jsx
 <HotTable
-  beforeColumnSort={function(currentSortConfig, destinationSortConfigs) {
-    const columnSortPlugin = this.getPlugin('columnSorting');
+  columnSorting={{
+    // implement your own comparator
+    compareFunctionFactory(sortOrder, columnMeta) {
+      return function(value, nextValue) {
+        // here, add a compare function
+        // that returns `-1`, or `0`, or `1`
+      },
+    },
+  }}
+/>
+```
 
-    columnSortPlugin.setSortConfig(destinationSortConfigs);
+:::
 
-    // const newData = ... // Calculated data set, ie. from an AJAX call.
+## Control sorting programmatically
```

Comment:
What developers want to do, programatically, with sorting:
- Enable / disable for the entire grid and particular columns
- Sort particular columns ascending or descending, or to restore the original state. That includes multi-column sorting.

Reply:
Ok, done